### DISCUSSION TITLE
mdast-util-definitions expected node

### DISCUSSION THREAD

**Discussion by PabloSzx** [syntax-tree/unist#63] [Q&A] [answered]
Is this a known issue trying to migrate from mdx v1 to mdx v2?

![image](https://user-images.githubusercontent.com/8672915/152447232-257549dc-eb7b-4355-899b-c31d56369e3d.png)

**Comment by ChristianMurphy** [comment]
@PabloSzx it likely is, try using the preview release of `next-mdx-remote` version 4 https://github.com/hashicorp/next-mdx-remote/releases/tag/v4.0.0-rc.1 which supports MDX 2.
Also make sure to update any mdast utils, hast utils, remark plugins, or rehype versions as well.

**Reply by PabloSzx** [reply]
Hello, all the available plugins were updated, and this was being tested on the-guild-docs, this is the draft PR https://github.com/the-guild-org/the-guild-docs/pull/473, this uses a fork of next-mdx-remote that has all the same main logic as the mdx v2 PR in next-mdx-remote https://github.com/hashicorp/next-mdx-remote/pull/211, so that shouldn't be the issue

**Reply by ChristianMurphy** [reply]
Next is known to be working with `@mdx-js/loader` version 2 https://codesandbox.io/s/github/mdx-js/.github/tree/main/sandbox-templates/mdx-loader-next
So I tend to think it is related to something in https://github.com/hashicorp/next-mdx-remote/pull/211

**Comment by PabloSzx** [comment, marked as answer]
I managed to narrow down the issues

Shiki integration fixed changing from https://github.com/stefanprobst/remark-shiki to https://github.com/stefanprobst/rehype-shiki

and https://github.com/elviswolcott/remark-admonitions no longer being maintained and finally stopped working with mdx v2
